src/rabbit_mgmt_util.erl

```diff
@@ -708,29 +709,70 @@ get_or_missing(K, L) ->
+        undefined -> node();
+        N         -> rabbit_nodes:make(
+                       binary_to_list(N))
+    end.
+
+direct_request(MethodName, Transformers, Extra, ErrorMsg, ReqData,
+               Context = #context{user = User}) ->
+    with_vhost_and_props(
+      fun(VHost, Props) ->
+              Method = props_to_method(MethodName, Props, Transformers, Extra),
+              Node = get_node(Props),
+              case rabbit_misc:rpc_call(Node, rabbit_channel, handle_method,
+                                        [Method, none, none,
+                                         VHost, User]) of
+                  {badrpc, nodedown} ->
+                      bad_request(
+                        list_to_binary(
+                          io_lib:format("Node ~s could not be contacted", [Node])),
+                        ReqData, Context);
+                  {badrpc, {'EXIT', #amqp_error{name = not_found, explanation = Explanation}}} ->
+                      not_found(Explanation, ReqData, Context);
+                  {badrpc, {'EXIT', #amqp_error{name = limit_reached, explanation = Explanation}}} ->
+                      rabbit_log:warning(ErrorMsg, [Explanation]),
+                      not_authorised(<<"Access refused.">>, ReqData, Context);
+                  {badrpc, {'EXIT', #amqp_error{name = access_refused, explanation = Explanation}}} ->
+                      rabbit_log:warning(ErrorMsg, [Explanation]),
+                      not_authorised(<<"Access refused.">>, ReqData, Context);
+                  {badrpc, {'EXIT', #amqp_error{name = not_allowed, explanation = Explanation}}} ->
+                      rabbit_log:warning(ErrorMsg, [Explanation]),
+                      not_authorised(<<"Access refused.">>, ReqData, Context);
+                  {badrpc, {'EXIT', #amqp_error{explanation = Explanation} = E}} ->
+                      rabbit_log:warning("AMQP ERROR is ~p~n", [E]),
+                      rabbit_log:warning(ErrorMsg, [Explanation]),
+                      bad_request(list_to_binary(Explanation), ReqData, Context);
+                  {badrpc, Reason} ->
+                      rabbit_log:warning(ErrorMsg, [Reason]),
+                      bad_request(
+                        list_to_binary(
+                          io_lib:format("Request to node ~s failed with ~p",
+                                        [Node, Reason])),
+                        ReqData, Context);
+                  _      -> {true, ReqData, Context}
+              end
+      end, ReqData, Context).
+
+with_vhost_and_props(Fun, ReqData, Context) ->
     case vhost(ReqData) of
         not_found ->
-            not_found(vhost_not_found, ReqData, Context);
+            not_found(rabbit_data_coercion:to_binary("vhost_not_found"),
+                      ReqData, Context);
         VHost ->
-            {ok, Body, ReqData1} = cowboy_req:body(ReqData),
+            {ok, Body, _ReqData} = cowboy_req:body(ReqData),
```

Comment:
I don't understand why the modified request state returned by `cowboy_req:body()` is ignored here.